types/react-redux/index.d.ts

```diff
@@ -173,11 +173,30 @@ interface Options extends ConnectOptions {
      * @default true
      */
     pure?: boolean;
+
+    /**
+     * When pure, compares incoming store state to its previous value.
+     * @default strictEqual
+     */
+    areStatesEqual?: (nextState: any, prevState: any) => boolean;
+    
+    /**
+     * When pure, compares incoming store state to its previous value.
+     * @default shallowEqual
+     */
+    areOwnPropsEqual?: (nextOwnProps: TOwnProps, prevOwnProps: TOwnProps) => boolean;
+    
+    /**
+     * When pure, compares the result of mapStateToProps to its previous value.
+     * @default shallowEqual
+     */
+    areStatePropsEqual?: (nextStateProps: TStateProps, prevStateProps: TStateProps) => boolean;
+
     /**
-     * If true, stores a ref to the wrapped component instance and makes it available via
-     * getWrappedInstance() method. Defaults to false.
+     * When pure, compares the result of mergeProps to its previous value.
+     * @default shallowEqual
      */
-    withRef?: boolean;
```

Comment:
looks like it should be moved inside of [`ConnectOptions`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments-1)